reark/src/main/java/io/reark/reark/data/stores/cores/ContentProviderStoreCoreBase.java

```diff
@@ -59,25 +67,73 @@
  * @param <U> Type of the data this store core contains.
  */
 public abstract class ContentProviderStoreCoreBase<U> {
-    private static final String TAG = ContentProviderStoreCoreBase.class.getSimpleName();
+
+    private final String TAG = getClass().getSimpleName();
 
     @NonNull
-    private final ContentResolver contentResolver;
+    private static final ContentProviderOperation NO_OPERATION = ContentProviderOperation.newInsert(Uri.EMPTY).build();
 
     @NonNull
-    private final ContentObserver contentObserver = getContentObserver();
+    private final ContentResolver contentResolver;
 
     @NonNull
     private final PublishSubject<Pair<U, Uri>> updateSubject = PublishSubject.create();
 
+    @NonNull
+    private final ObjectLockHandler<Uri> locker = new ObjectLockHandler<>();
+
+    @Nullable
+    private Subscription updateSubscription;
```

Comment:
Our stores are singletons that don't really get disposed. This subscription is also basically just never going out. Do you think that the stores should follow some lifecycle too?